generatedocs.py

```diff
@@ -0,0 +1,413 @@
+    # If there were any matches to regex, set the content as the example
+    if len(lines) > 0:
+        return { "header" : header, "description": description, "content" : content, "lines" : lines }
+    else: 
+        return None
+
+# Takes an asset component name and matches it to all files in a 
+# path, and return the shortest asset with matches.
+def findShortestAssetInPath(path, name):
+    assetFiles = assetsInPathRecursive(path)
+    # Sort by shortest asset file first                
+    # This will ensure the simplest asset is displayed
+    assetFiles.sort(key=getFileLength)
+
+    # Find example for the asset component
+    for assetFile in assetFiles:   
+        # Search for Type = "<name>"
+        regex = r'Type = \"' + name + r'\"'
+        example = getLinesAndContentFromFile(assetFile, regex)
+        # As soon as we have a match, we return it
+        # It will be the shortest assets since they are sorted on length
+        if example:
+            return example
+    # If nothing found, return None
+    return None
+
+# Search through all example assets for the component name
+# Returns file content and line numbers for where the name occurs
+def findAssetExample(assetsFolder, category, name):
+    examplesFolder = os.path.join(assetsFolder, "examples")
+    
+    # Search pass 1: look up folder “assets/<category>/<assetcomponentname>/”
+    # and see if it exists. If it does, add all files in that folder (no subdirectories)
+    assetDirectory = os.path.join(examplesFolder, (os.path.join(category, name)).lower())
+    if os.path.exists(assetDirectory):
+        filenames = assetsInPath(assetDirectory)
+        examples = []
+        for filename in filenames:
+            # We search for the asset component <name> or "<name>"
+            regex = r'\b' + name + r'\b|\b\"' + name + r'\"\b'
+            example = getLinesAndContentFromFile(filename, regex, True)
+            examples.append(example)
+        return examples
+
+    # Search pass 2: search through all assets in the **examples** folder and add the 
+    # shortest asset
+    example = findShortestAssetInPath(examplesFolder, name)
+    if example:
+        return [example]
+    
+    # Search pass 3: search through all assets in the **assets** folder and add the 
+    # shortest asset
+    example = findShortestAssetInPath(assetsFolder, name)
+    if example:
+        return [example]
+    
+    # If nothing found, return empty array
+    return []
+
+# Group members by optionality while preserving the alphabetical order
+def groupMembersByOptionality(members):
+    nonOptionalIndices = []
+    optionalIndices = []
+    for i, member in enumerate(members):
+        if member["optional"]:
+            optionalIndices.append(i)
+        else:
+            nonOptionalIndices.append(i)
+    indices = nonOptionalIndices + optionalIndices
+    return [members[i] for i in indices]
+
+def findAssetScreenshot(name):
```

Comment:
Since all the other funciton have a describing comment, maybe this one should too?